### PR TITLE
ibus: 1.5.28 -> 1.5.29

### PR DESCRIPTION
**Pull request by oliviacrain**
## Description of changes

https://github.com/ibus/ibus/releases/tag/1.5.29
https://github.com/ibus/ibus/releases/tag/1.5.29-rc1

This update is motivated by upstream changes that make this package build reproducibly when using parallel `make`.

See: #230290

Changes:
- Adds libdbusmenu-gtk3 dependency, now required for appindicator support
- Adds workaround for interaction between `AX_PROG_{CC,CXX}` autoconf macro and unwrapped GCC binaries (see #21751)
- Patches generated version number to remove incorrect `-rcX` suffix
- Removes upstreamed patches, rebases `build-without-dbus-launch.patch`
- Formatted with `nixpkgs-fmt`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>gnomeExtensions.easyScreenCast</li>
  </ul>
</details>
<details>
  <summary>82 packages built:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>budgie.budgie-control-center</li>
    <li>budgie.budgie-control-center.debug</li>
    <li>budgie.budgie-desktop</li>
    <li>budgie.budgie-desktop-with-plugins</li>
    <li>budgie.budgie-gsettings-overrides</li>
    <li>budgiePlugins.budgie-analogue-clock-applet</li>
    <li>budgiePlugins.budgie-user-indicator-redux</li>
    <li>cinnamon.cinnamon-gsettings-overrides</li>
    <li>emacsPackages.ac-mozc</li>
    <li>emacsPackages.mozc</li>
    <li>emacsPackages.mozc-cand-posframe</li>
    <li>emacsPackages.mozc-im</li>
    <li>emacsPackages.mozc-popup</li>
    <li>emacsPackages.mozc-temp</li>
    <li>enlightenment.econnman</li>
    <li>enlightenment.ecrire</li>
    <li>enlightenment.efl</li>
    <li>enlightenment.enlightenment</li>
    <li>enlightenment.ephoto</li>
    <li>enlightenment.evisum</li>
    <li>enlightenment.rage</li>
    <li>enlightenment.terminology</li>
    <li>gnome-browser-connector</li>
    <li>gnome.gnome-control-center</li>
    <li>gnome.gnome-control-center.debug</li>
    <li>gnome.gnome-flashback</li>
    <li>gnome.gnome-panel-with-modules</li>
    <li>gnome.gnome-session</li>
    <li>gnome.gnome-session.debug</li>
    <li>gnome.gnome-session.sessions</li>
    <li>gnome.gnome-shell</li>
    <li>gnome.gnome-shell.debug</li>
    <li>gnome.gnome-shell.devdoc</li>
    <li>gnome.gnome-terminal</li>
    <li>gnome.gnome-tweaks</li>
    <li>gnome.nixos-gsettings-overrides</li>
    <li>gnomeExtensions.gsconnect</li>
    <li>gnomeExtensions.gsconnect.installedTests</li>
    <li>ibus</li>
    <li>ibus-engines.anthy</li>
    <li>ibus-engines.hangul</li>
    <li>ibus-engines.kkc</li>
    <li>ibus-engines.libpinyin</li>
    <li>ibus-engines.libthai</li>
    <li>ibus-engines.m17n</li>
    <li>ibus-engines.mozc</li>
    <li>ibus-engines.openbangla-keyboard</li>
    <li>ibus-engines.rime</li>
    <li>ibus-engines.table</li>
    <li>ibus-engines.table-chinese</li>
    <li>ibus-engines.table-others</li>
    <li>ibus-engines.typing-booster</li>
    <li>ibus-engines.typing-booster-unwrapped</li>
    <li>ibus-engines.uniemoji</li>
    <li>ibus-with-plugins</li>
    <li>ibus.dev</li>
    <li>ibus.installedTests</li>
    <li>latte-dock</li>
    <li>libsForQt5.kdeplasma-addons</li>
    <li>libsForQt5.plasma-desktop</li>
    <li>mlterm</li>
    <li>mlterm-wayland</li>
    <li>mojave-gtk-theme</li>
    <li>pantheon.elementary-greeter</li>
    <li>pantheon.elementary-session-settings</li>
    <li>pantheon.switchboard-plug-keyboard</li>
    <li>pantheon.switchboard-with-plugs</li>
    <li>pantheon.wingpanel-applications-menu</li>
    <li>pantheon.wingpanel-indicator-keyboard</li>
    <li>pantheon.wingpanel-with-indicators</li>
    <li>phosh</li>
    <li>phosh-mobile-settings</li>
    <li>polychromatic</li>
    <li>python310Packages.pythonefl</li>
    <li>python310Packages.pythonefl.dist</li>
    <li>python311Packages.pythonefl</li>
    <li>python311Packages.pythonefl.dist</li>
    <li>snippetexpanderx</li>
    <li>snippetpixie</li>
    <li>vimix-gtk-themes</li>
    <li>whitesur-gtk-theme</li>
  </ul>
</details>

Result of `nixpkgs-review` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>gnomeExtensions.easyScreenCast</li>
    <li>ibus-engines.openbangla-keyboard</li>
  </ul>
</details>
<details>
  <summary>81 packages built:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>budgie.budgie-control-center</li>
    <li>budgie.budgie-control-center.debug</li>
    <li>budgie.budgie-desktop</li>
    <li>budgie.budgie-desktop-with-plugins</li>
    <li>budgie.budgie-gsettings-overrides</li>
    <li>budgiePlugins.budgie-analogue-clock-applet</li>
    <li>budgiePlugins.budgie-user-indicator-redux</li>
    <li>cinnamon.cinnamon-gsettings-overrides</li>
    <li>emacsPackages.ac-mozc</li>
    <li>emacsPackages.mozc</li>
    <li>emacsPackages.mozc-cand-posframe</li>
    <li>emacsPackages.mozc-im</li>
    <li>emacsPackages.mozc-popup</li>
    <li>emacsPackages.mozc-temp</li>
    <li>enlightenment.econnman</li>
    <li>enlightenment.ecrire</li>
    <li>enlightenment.efl</li>
    <li>enlightenment.enlightenment</li>
    <li>enlightenment.ephoto</li>
    <li>enlightenment.evisum</li>
    <li>enlightenment.rage</li>
    <li>enlightenment.terminology</li>
    <li>gnome-browser-connector</li>
    <li>gnome.gnome-control-center</li>
    <li>gnome.gnome-control-center.debug</li>
    <li>gnome.gnome-flashback</li>
    <li>gnome.gnome-panel-with-modules</li>
    <li>gnome.gnome-session</li>
    <li>gnome.gnome-session.debug</li>
    <li>gnome.gnome-session.sessions</li>
    <li>gnome.gnome-shell</li>
    <li>gnome.gnome-shell.debug</li>
    <li>gnome.gnome-shell.devdoc</li>
    <li>gnome.gnome-terminal</li>
    <li>gnome.gnome-tweaks</li>
    <li>gnome.nixos-gsettings-overrides</li>
    <li>gnomeExtensions.gsconnect</li>
    <li>gnomeExtensions.gsconnect.installedTests</li>
    <li>ibus</li>
    <li>ibus-engines.anthy</li>
    <li>ibus-engines.hangul</li>
    <li>ibus-engines.kkc</li>
    <li>ibus-engines.libpinyin</li>
    <li>ibus-engines.libthai</li>
    <li>ibus-engines.m17n</li>
    <li>ibus-engines.mozc</li>
    <li>ibus-engines.rime</li>
    <li>ibus-engines.table</li>
    <li>ibus-engines.table-chinese</li>
    <li>ibus-engines.table-others</li>
    <li>ibus-engines.typing-booster</li>
    <li>ibus-engines.typing-booster-unwrapped</li>
    <li>ibus-engines.uniemoji</li>
    <li>ibus-with-plugins</li>
    <li>ibus.dev</li>
    <li>ibus.installedTests</li>
    <li>latte-dock</li>
    <li>libsForQt5.kdeplasma-addons</li>
    <li>libsForQt5.plasma-desktop</li>
    <li>mlterm</li>
    <li>mlterm-wayland</li>
    <li>mojave-gtk-theme</li>
    <li>pantheon.elementary-greeter</li>
    <li>pantheon.elementary-session-settings</li>
    <li>pantheon.switchboard-plug-keyboard</li>
    <li>pantheon.switchboard-with-plugs</li>
    <li>pantheon.wingpanel-applications-menu</li>
    <li>pantheon.wingpanel-indicator-keyboard</li>
    <li>pantheon.wingpanel-with-indicators</li>
    <li>phosh</li>
    <li>phosh-mobile-settings</li>
    <li>polychromatic</li>
    <li>python310Packages.pythonefl</li>
    <li>python310Packages.pythonefl.dist</li>
    <li>python311Packages.pythonefl</li>
    <li>python311Packages.pythonefl.dist</li>
    <li>snippetexpanderx</li>
    <li>snippetpixie</li>
    <li>vimix-gtk-themes</li>
    <li>whitesur-gtk-theme</li>
  </ul>
</details>

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by oliviacrain** [comment]
cc @yanateras, looks like the maintainer list has an incorrect account name for you!

**Comment by raboof** [comment]
Result of `nixpkgs-review pr 277423` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>gnomeExtensions.easyScreenCast</li>
  </ul>
</details>
<details>
  <summary>82 packages built:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>budgie.budgie-control-center</li>
    <li>budgie.budgie-control-center.debug</li>
    <li>budgie.budgie-desktop</li>
    <li>budgie.budgie-desktop-with-plugins</li>
    <li>budgie.budgie-gsettings-overrides</li>
    <li>budgiePlugins.budgie-analogue-clock-applet</li>
    <li>budgiePlugins.budgie-user-indicator-redux</li>
    <li>cinnamon.cinnamon-gsettings-overrides</li>
    <li>emacsPackages.ac-mozc</li>
    <li>emacsPackages.mozc</li>
    <li>emacsPackages.mozc-cand-posframe</li>
    <li>emacsPackages.mozc-im</li>
    <li>emacsPackages.mozc-popup</li>
    <li>emacsPackages.mozc-temp</li>
    <li>enlightenment.econnman</li>
    <li>enlightenment.ecrire</li>
    <li>enlightenment.efl</li>
    <li>enlightenment.enlightenment</li>
    <li>enlightenment.ephoto</li>
    <li>enlightenment.evisum</li>
    <li>enlightenment.rage</li>
    <li>enlightenment.terminology</li>
    <li>gnome-browser-connector</li>
    <li>gnome.gnome-control-center</li>
    <li>gnome.gnome-control-center.debug</li>
    <li>gnome.gnome-flashback</li>
    <li>gnome.gnome-panel-with-modules</li>
    <li>gnome.gnome-session</li>
    <li>gnome.gnome-session.debug</li>
    <li>gnome.gnome-session.sessions</li>
    <li>gnome.gnome-shell</li>
    <li>gnome.gnome-shell.debug</li>
    <li>gnome.gnome-shell.devdoc</li>
    <li>gnome.gnome-terminal</li>
    <li>gnome.gnome-tweaks</li>
    <li>gnome.nixos-gsettings-overrides</li>
    <li>gnomeExtensions.gsconnect</li>
    <li>gnomeExtensions.gsconnect.installedTests</li>
    <li>ibus</li>
    <li>ibus-engines.anthy</li>
    <li>ibus-engines.hangul</li>
    <li>ibus-engines.kkc</li>
    <li>ibus-engines.libpinyin</li>
    <li>ibus-engines.libthai</li>
    <li>ibus-engines.m17n</li>
    <li>ibus-engines.mozc</li>
    <li>ibus-engines.openbangla-keyboard</li>
    <li>ibus-engines.rime</li>
    <li>ibus-engines.table</li>
    <li>ibus-engines.table-chinese</li>
    <li>ibus-engines.table-others</li>
    <li>ibus-engines.typing-booster</li>
    <li>ibus-engines.typing-booster-unwrapped</li>
    <li>ibus-engines.uniemoji</li>
    <li>ibus-with-plugins</li>
    <li>ibus.dev</li>
    <li>ibus.installedTests</li>
    <li>latte-dock</li>
    <li>libsForQt5.kdeplasma-addons (plasma5Packages.kdeplasma-addons)</li>
    <li>libsForQt5.plasma-desktop (plasma5Packages.plasma-desktop)</li>
    <li>mlterm</li>
    <li>mlterm-wayland</li>
    <li>mojave-gtk-theme</li>
    <li>pantheon.elementary-greeter</li>
    <li>pantheon.elementary-session-settings</li>
    <li>pantheon.switchboard-plug-keyboard</li>
    <li>pantheon.switchboard-with-plugs</li>
    <li>pantheon.wingpanel-applications-menu</li>
    <li>pantheon.wingpanel-indicator-keyboard</li>
    <li>pantheon.wingpanel-with-indicators</li>
    <li>phosh</li>
    <li>phosh-mobile-settings</li>
    <li>polychromatic</li>
    <li>python310Packages.pythonefl</li>
    <li>python310Packages.pythonefl.dist</li>
    <li>python311Packages.pythonefl</li>
    <li>python311Packages.pythonefl.dist</li>
    <li>snippetexpanderx</li>
    <li>snippetpixie</li>
    <li>vimix-gtk-themes</li>
    <li>whitesur-gtk-theme</li>
  </ul>
</details>

**Review by raboof — APPROVED** [review]
Looks OK to me, but I'll hold off merging to give maintainers @ttuegel and @yanateras a chance to have a look.

**Comment by yanateras** [comment]
On it.

**Comment by raboof** [comment]
@yanateras do you want us to wait until you've had time to have a look?

File: pkgs/tools/inputmethods/ibus/default.nix
Comment (jtojnar) [inline pkgs/tools/inputmethods/ibus/default.nix:149]: The upstream for this dependency is [basically dead](https://launchpad.net/libdbusmenu) and in the past we wanted to [get rid of it](https://github.com/NixOS/nixpkgs/issues/96420) in favour of Debian forks. But it looks like that is now dead as well: https://github.com/AyatanaIndicators/libdbusmenu

File: pkgs/tools/inputmethods/ibus/default.nix
Comment (jtojnar) [inline pkgs/tools/inputmethods/ibus/default.nix:1]: Looks like this version [enabled Wayland by default](https://github.com/ibus/ibus/commit/494448b78511e306089ea172993910295e654219). We should do it as well.

Comment (jtojnar) [inline pkgs/tools/inputmethods/ibus/default.nix:1]: It no longer builds for me when rebased onto `nixos-unstable`:

```
emojier.c:9292:1: error: unknown type name 'pointer'; did you mean 'gpointer'?
 9292 | pointer*) array)[length]) {
      | ^~~~~~~
      | gpointer
emojier.c:9292:9: error: expected identifier or '(' before ')' token
 9292 | pointer*) array)[length]) {
      |         ^
emojier.c:9295:9: error: expected identifier or '(' before '}' token
 9295 |         }
      |         ^
emojier.c:9296:9: error: expected identifier or '(' before 'return'
 9296 |         return length;
      |         ^~~~~~
emojier.c:9297:1: error: expected identifier or '(' before '}' token
 9297 | }
```

**Review by jtojnar — CHANGES_REQUESTED** [review]

File: pkgs/tools/inputmethods/ibus/default.nix
Comment (oliviacrain) [reply pkgs/tools/inputmethods/ibus/default.nix:1]: Done, thanks!

File: pkgs/tools/inputmethods/ibus/default.nix
Comment (oliviacrain) [reply pkgs/tools/inputmethods/ibus/default.nix:148]: Is there a course of action you'd suggest I take here? I hate to add new dependencies on dead libraries, but I'd also hate to regress on appindicator support.

Comment (jtojnar) [reply pkgs/tools/inputmethods/ibus/default.nix:148]: Looking at https://github.com/ibus/ibus/commit/8923456a5d924c640e70e3755ea60aea84246910, I guess we have to keep it for now.

**Review by jtojnar — APPROVED** [review]

**Pull request merged**
